app/views/shared/_sidebar.html.erb

```diff
@@ -1,195 +1,208 @@
+      <%= t('categories.sesam.plural') %>
+    </a>
   </li>
-  <li class="nav-item">
-    <% if !lecture || !lecture.kiwi?(current_user) %>
-      <span class="nav-link disabled">
-        <%= t('categories.kiwi.singular') %>
-      </span>
-    <% else %>
-      <%= link_to t('categories.kiwi.singular'),
-                  lecture_food_path(lecture,
-                                    params: { project: 'kiwi' }),
-                  class: 'nav-link' %>
-    <% end %>
+
+  <%# Examples (Erdbeere) %>
+  <% path = show_structures_path(lecture) %>
+  <% active_class = get_class_for_path(path) %>
+  <li class="sidebar-item nav-link <%= active_class %>
+          <%= !lecture || !lecture.structure_ids.any? ? "disabled" : ""%>">
+    <a href="<%= url_for(path)%>">
+      <i class="bi bi-box<%= active_class.present? ? "-fill" : "" %>"></i>
+      <%= t('categories.erdbeere.plural') %>
+    </a>
   </li>
-  <li class="nav-item">
-    <% if !lecture || !lecture.reste?(current_user) %>
-      <span class="nav-link disabled">
-        <%= t('categories.reste.singular') %>
-      </span>
-    <% else %>
-      <%= link_to t('categories.reste.singular'),
-                  lecture_food_path(lecture,
-                                    params: { project: 'reste' }),
-                  class: 'nav-link' %>
-    <% end %>
+
+  <%# Repetition (Kiwi) %>
+  <% project_name = 'kiwi' %>
+  <% fullpath = lecture_food_path(lecture, params: { project: project_name })%>
+  <% active_class = get_class_for_project(project_name) %>
+  <li class="sidebar-item nav-link <%= active_class %>
+          <%= !lecture || !lecture.kiwi?(current_user) ? "disabled" : ""%>">
+    <a href="<%= url_for(fullpath)%>">
+      <i class="bi bi-layers<%= active_class.present? ? "-fill" : "" %>"></i>
+      <%= t('categories.kiwi.singular') %>
+    </a>
   </li>
+
+  <%# Diverse (Rest) %>
+  <% project_name = 'reste' %>
+  <% fullpath = lecture_food_path(lecture, params: { project: project_name })%>
+  <% active_class = get_class_for_project(project_name) %>
+  <li class="sidebar-item nav-link <%= active_class %>
+          <%= !lecture || !lecture.reste?(current_user) ? "disabled" : ""%>">
+    <a href="<%= url_for(fullpath)%>">
+      <i class="bi bi-circle<%= active_class.present? ? "-fill" : "" %>"></i>
+      <%= t('categories.reste.singular') %>
+    </a>
+  </li>
+
+
   <% if lecture %>
-    <span class="border-top my-3">
-    </span>
-    <li class="nav-item">
-      <% if lecture.organizational %>
-        <%= link_to t('categories.organizational_html'),
-                    organizational_path(lecture),
-                    class: 'nav-link' %>
-      <% else %>
-        <span class="nav-link disabled">
-          <%= t('categories.organizational_html') %>
-        </span>
-      <% end %>
+    <%# Divider %>
+    <span class="border-top my-3"></span>
+
+    <%# Forum %>
+    <% path = "/forum/" + lecture.forum.slug %>
```

Comment:
One error we get right now:

Log in as teacher and click on helpdesk. This will show up:

<img src="https://user-images.githubusercontent.com/37160523/230678904-dcfde430-69fb-46bd-932d-26f0f582a687.png" width="500px"/>

Apparently, not every lecture forum model has a slug. What would be the best option here to avoid this error? Test if slug is present and if not disable the forum button?

Reply:
In production, there are a lot of lectures that do not have forum (some teachers just do not want them). So, it should be tested whether there is a forum using `lecture.forum?`.